core/src/main/java/io/aiven/klaw/service/SchemaRegistrySyncControllerService.java

```diff
@@ -0,0 +1,320 @@
+        schemaSubjectInfoResponse.setSchemaVersions(schemaInfoOfTopic.getSchemaVersions());
+        schemaSubjectInfoResponse.setTopic(schemaInfoOfTopic.getTopic());
+        schemaSubjectInfoResponseList.add(schemaSubjectInfoResponse);
+      }
+
+      schemaSubjectInfoResponseList =
+          filterTopicsNotInDb(
+              schemaSubjectInfoResponseList, topicsFromSOT, schemaEnvSelected.getId(), tenantId);
+
+      if (!showAllTopics) {
+        schemaSubjectInfoResponseList =
+            schemaSubjectInfoResponseList.stream()
+                .filter(schema -> schema.getRemarks().equals(NOT_IN_SYNC))
+                .toList();
+      }
+
+      syncSchemasList.setSchemaSubjectInfoResponseList(
+          getPagedResponse(pageNo, currentPage, schemaSubjectInfoResponseList, tenantId));
+      return syncSchemasList;
+    } catch (Exception e) {
+      log.error("Exception:", e);
+      throw new KlawException(e.getMessage());
+    }
+  }
+
+  private List<SchemaSubjectInfoResponse> filterTopicsNotInDb(
+      List<SchemaSubjectInfoResponse> schemaSubjectInfoResponseList,
+      List<Topic> topicsFromSOT,
+      String schemaEnvId,
+      int tenantId) {
+    List<SchemaSubjectInfoResponse> updatedList = new ArrayList<>();
+    Map<String, Set<String>> topicSchemaVersionsInDb =
+        manageDatabase
+            .getHandleDbRequests()
+            .getTopicAndVersionsForEnvAndTenantId(schemaEnvId, tenantId);
+    for (SchemaSubjectInfoResponse schemaSubjectInfoResponse : schemaSubjectInfoResponseList) {
+      Optional<Topic> optionalTopic =
+          topicsFromSOT.stream()
+              .filter(t -> t.getTopicname().equals(schemaSubjectInfoResponse.getTopic()))
+              .findAny();
+      if (optionalTopic.isPresent()) {
+        schemaSubjectInfoResponse.setTeamId(optionalTopic.get().getTeamId());
+        validateSchemas(
+            schemaSubjectInfoResponse,
+            topicSchemaVersionsInDb.get(schemaSubjectInfoResponse.getTopic()));
+        updatedList.add(schemaSubjectInfoResponse);
+      }
+    }
+    return updatedList;
+  }
+
+  private void validateSchemas(
+      SchemaSubjectInfoResponse schemaSubjectInfoResponse, Set<String> schemaVersionsOnDb) {
+    Set<Integer> schemaVersionsOnCluster = schemaSubjectInfoResponse.getSchemaVersions();
+
+    if (schemaVersionsOnDb == null) {
+      schemaSubjectInfoResponse.setRemarks(NOT_IN_SYNC);
+      return;
+    }
+
+    if (schemaVersionsOnCluster.size() != schemaVersionsOnDb.size()) {
+      schemaSubjectInfoResponse.setRemarks(NOT_IN_SYNC);
+      return;
+    } else {
+      schemaSubjectInfoResponse.setRemarks(IN_SYNC);
+    }
+
+    schemaVersionsOnCluster.forEach(
+        ver -> {
+          if (!schemaVersionsOnDb.contains(ver + "")) {
+            schemaSubjectInfoResponse.setRemarks(NOT_IN_SYNC);
+          }
+        });
+  }
+
+  private List<SchemaSubjectInfoResponse> getPagedResponse(
+      String pageNo,
```

Comment:
It doesn't need to be done here but we should definitely have a single source of truth for paging responses i'll open an issue